File: voldemort/python/read/voldemort_read.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	voldemort_read.py
#
#					Jan/27/2012
import voldemort
import json
import sys
#
sys.path.append ('/var/www/data_base/common/python_common')
from text_manipulate import dict_display_proc
from voldemort_manipulate import voldemort_to_dict_proc
# ------------------------------------------------------------
print	("*** 開始 ***")
#
client = voldemort.StoreClient('cities', [('localhost', 6666)])

dict_aa = voldemort_to_dict_proc (client)
dict_display_proc (dict_aa)
#
print	("*** 終了 ***")
# ------------------------------------------------------------

File: voldemort/jruby/read/voldemort_read.rb
#! /usr/bin/jruby
# -*- coding: utf-8 -*-
#
#	voldemort_read.rb
#
#					Sep/09/2013
#
#include	Java
import	java.util.HashMap
import	voldemort.client.SocketStoreClientFactory
import	voldemort.client.ClientConfig
import	voldemort.versioning.Versioned
#
#require 'rubygems'
#require 'open-uri'
#require 'json'
#
# ---------------------------------------------------------------------
#
puts	"*** 開始 ***"
#
factory = SocketStoreClientFactory(ClientConfig().setBootstrapUrls("tcp://localhost:6666"))
client = factory.getStoreClient("cities")
#
data_aa['keys'].each {|key|
	url_city = url_base + '/' + key
	response= open(url_city,:proxy=>nil)
	str_city=response.read
	city=JSON.parse(str_city)
	print key,"\t"
	print city["name"],"\t"
	print city["population"],"\t"
	print city["date_mod"],"\n"
	}
#
puts	"*** 終了 ***"
#
# ---------------------------------------------------------------------
